lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart

```diff
@@ -69,8 +69,7 @@ final class ViewFocusBinding {
   });
 
   late final DomEventListener _handleKeyDown = createDomEventListener((DomEvent event) {
-    event as DomKeyboardEvent;
-    if (event.shiftKey ?? false) {
+    if (event is DomKeyboardEvent && event.shiftKey == true) {
```

Comment:
Is there a way to explain how it is possible to get a non-`DomKeyboardEvent` in a listener that exclusively listens to "keydown" events? Maybe an MDN article or a StackOverflow thread?

Reply:
Also, is there a way to test this?

Reply:
I couldn't find anything concrete from my 10 minutes search. I couldn't reproduce this issue in firefox tho so wondering if this is a chrome/chromium only issue.

Reply:
Can you please add a comment above that links to the issue and explains that the non-KeyDown events come from Chrome autofill?

Reply:
@yjbanov Chrome is sending a synthetic Event of type "keydown" when autofill completes that is lacking `shiftKey` (and pretty much all the properties from a normal keydown). Compare:

| Normal (pressed "P") | Autofill |
|--------|----------|
| <img width="416" alt="Screenshot 2024-07-11 at 1 32 00 PM" src="https://github.com/user-attachments/assets/a73b3a71-6961-40ff-bba1-6081b6fd7db5"> | <img width="394" alt="Screenshot 2024-07-11 at 1 32 15 PM" src="https://github.com/user-attachments/assets/3a09de38-3cf2-4ffc-86a5-e92228d56468"> |

(Test it [here](https://jsfiddle.net/ditman/jg637yfs/))

Apologies @tugorez, I thought that making the property nullable would work (and that JS-interop would just "cast" across types, but it seems it doesn't :S

Reply:
> Apologies @tugorez, I thought that making the property nullable would work (and that JS-interop would just "cast" across types, but it seems it doesn't :S

Hey no worries, FWIW chrome is the one doing funky stuff. It is firing a totally different event type `Event`. This feels like a bug in chrome (I couldn't reproduce it in  Firefox).

Reply:
(PS: I do think this is a Chrome bug 😄)